homeassistant/components/nanoleaf/__init__.py

```diff
@@ -80,8 +80,15 @@ async def touch_event_callback(event: TouchEvent) -> None:
         )
     )
 
+    async def _cancel_listener() -> None:
+        event_listener.cancel()
+        with suppress(RuntimeError):
```

Comment:
I think you want asyncio.CancelledError here since this is going to consume exceptions other than the one we know is expected